coral-spark/src/main/java/com/linkedin/coral/spark/IRRelToSparkRelTransformer.java

```diff
@@ -368,6 +364,29 @@ else if (call.getOperands().size() == 2) {
       return Optional.empty();
     }
 
+    // Calcite entails the nullability of an expression by casting it to the correct nullable type.
+    // However, for complex types like ARRAY<STRING NOT NULL> (element non-nullable, but top-level nullable),
+    // the translated SQL will still be `CAST(XXX) AS ARRAY<STRING>`. Since Spark treats a Cast target type
+    // as always nullable (both inner and outer), it will treat the SQL type as ARRAY<STRING>, this deviates
+    // from the nullability represented in RelNode/Coral-Schema.
```

Comment:
Could you clarify the nullability of RelNode/Coral-Schema here? Maybe point to the unit test you added in Coral-Schema.

Reply:
added